lib/http2Response.js

```diff
@@ -0,0 +1,40 @@
+/*!
+ * express
+ * Copyright(c) 2009-2013 TJ Holowaychuk
+ * Copyright(c) 2014-2015 Douglas Christopher Wilson
+ * MIT Licensed
+ */
+
+'use strict';
+
+/**
+ * Module dependencies.
+ * @private
+ */
+var Http2ServerResponse = require('http2').Http2ServerResponse;
+var decorator = require('./responseDecorator');
+/**
+ * Response prototype.
+ * @public
+ */
+
+var res = decorator(Object.create(Http2ServerResponse.prototype));
+
+/**
+ * Add setter for statusMessage as it's deprecated, node doesn't set setter only getter
+ * so without setting this setter it fails
+ * Remove if node adds deprecation setter as well
+ */
+Object.defineProperty(res, 'statusMessage', {
+  set: function (x) {

```

Comment:
I had no idea you can define a setter separately from a getter, but apparently this works. 

This can be a `set() {}` by the way (shorthand property)

Reply:
Nevermind, it can't, I forgot this needs to run on old Node since it's not versioned with core. Sorry for the misleading comment!

Reply:
We should just wait for the setter to be in Node.js core, as the wrapper claims to be compatible for you only use the public API and setting the statusMessage is public API.

Reply:
I will try to see what to do about the status message on the node part then.

However one thing I noticed is that express always tries to set
statusMessage which if node add setter will result in deprecation message
if you use express with http2 - any ideas how to avoid that?

Reply:
I would say just change where it is being set to base that on the http version property of the request.

Reply:
The change to Node.js core add a setter looks like it will land very soon (PR https://github.com/nodejs/node/pull/15193) so we won't nee the custom setter any longer.

Reply:
Awesome, will remove it then.

Reply:
Landed in [Version 8.5.0](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V8.md#2017-09-12-version-850-current-mylesborins)